app/src/androidTest/java/com/github/se/project/ui/overview/OverViewTest.kt

```diff
@@ -1,363 +1,398 @@
+      mutableListOf(
+          Lesson(
+              id = "1",
+              title = "Physics Tutoring",
+              description = "Mechanics and Thermodynamics",
+              subject = Subject.PHYSICS,
+              languages = listOf(Language.ENGLISH),
+              tutorUid = listOf("tutor"),
+              studentUid = "student",
+              minPrice = 20.0,
+              maxPrice = 40.0,
+              timeSlot = "2024-10-10T10:00:00",
+              status = LessonStatus.STUDENT_REQUESTED,
+              latitude = 0.0,
+              longitude = 0.0),
+          Lesson(
+              id = "2",
+              title = "Math Tutoring",
+              description = "Algebra and Calculus",
+              subject = Subject.ANALYSIS,
+              languages = listOf(Language.ENGLISH),
+              tutorUid = listOf("tutor"),
+              studentUid = "student",
+              minPrice = 20.0,
+              maxPrice = 40.0,
+              timeSlot = "2024-10-10T11:00:00",
+              status = LessonStatus.STUDENT_REQUESTED,
+              latitude = 0.0,
+              longitude = 0.0))
+
+  private val instantLesson =
+      Lesson(
+          id = "3",
+          title = "Instant ICC Tutoring",
+          description = "Algebra and Calculus",
+          subject = Subject.ICC,
+          languages = listOf(Language.ENGLISH),
+          tutorUid = listOf("tutor123"),
+          studentUid = "student123",
+          minPrice = 20.0,
+          maxPrice = 40.0,
+          timeSlot = "2024-10-10Tinstant",
+          status = LessonStatus.INSTANT_CONFIRMED,
+          latitude = 0.0,
+          longitude = 0.0)
+
+  private val mockstudentProfileFlow = MutableStateFlow<Profile?>(tutorProfile)
+
+  private val cancelledLesson =
+      listOf(
+          Lesson(
+              id = "3",
+              title = "Math Tutoring",
+              description = "Algebra and Calculus",
+              subject = Subject.ANALYSIS,
+              languages = listOf(Language.ENGLISH),
+              tutorUid = listOf("tutor"),
+              studentUid = "student",
+              minPrice = 20.0,
+              maxPrice = 40.0,
+              price = 25.0,
+              timeSlot = "2024-10-10T11:00:00",
+              status = LessonStatus.STUDENT_CANCELLED,
+              latitude = 0.0,
+              longitude = 0.0))
+
+  private val mockProfileFlow = MutableStateFlow<Profile?>(tutorProfile)
+
+  private val currentUserLessonsFlow = MutableStateFlow(mockLessons)
+
+  private val cancelledLessonsFlow = MutableStateFlow<List<Lesson>>(listOf())
+
+  @Before
+  fun setup() {
+    /// Mock the dependencies
+    profilesRepository = mock(ProfilesRepository::class.java)
+    lessonRepository = mock(LessonRepository::class.java)
+    navController = mock()
+
+    // Initialize the ViewModel with a spy
+    listProfilesViewModel = ListProfilesViewModel(profilesRepository)
+    listProfilesViewModel = spy(listProfilesViewModel)
```

Comment:
Very good idea to use a spy! Definitely a feature of Mockito we have underutilised